core/embed/rust/src/ui/model_tt/component/homescreen.rs

```diff
@@ -0,0 +1,266 @@
+        }
+    }
+
+    fn paint_notification(&self) {
+        if !self.usb_connected {
+            let (color, icon) = Self::level_to_style(0);
+            NotificationFrame::<Empty, T>::paint_notification(
+                AREA,
+                icon,
+                "NO USB CONNECTION",
+                color,
+            );
+        } else if let Some((notification, level)) = &self.notification {
+            let (color, icon) = Self::level_to_style(*level);
+            NotificationFrame::<Empty, T>::paint_notification(
+                AREA,
+                icon,
+                notification.as_ref(),
+                color,
+            );
+        }
+    }
+
+    fn paint_loader(&mut self) {
+        let top_center = AREA.top_left().center(AREA.top_right());
+        display::text_center(
+            top_center + Offset::y(HOLD_Y),
+            "HOLD TO LOCK",
+            Font::BOLD,
+            theme::FG,
+            theme::BG,
+        );
+        self.loader.paint()
+    }
+
+    fn event_usb(&mut self, ctx: &mut EventCtx, event: Event) {
+        if let Event::USB(USBEvent::Connected(is_connected)) = event {
+            if self.usb_connected != is_connected {
+                self.usb_connected = is_connected;
+                ctx.request_paint();
+            }
+        }
+    }
+
+    fn event_hold(&mut self, ctx: &mut EventCtx, event: Event) -> bool {
+        match event {
+            Event::Touch(TouchEvent::TouchStart(_)) => {
+                if self.loader.is_animating() {
+                    self.loader.start_growing(ctx, Instant::now());
+                } else {
+                    self.delay = Some(ctx.request_timer(LOADER_DELAY));
+                }
+            }
+            Event::Touch(TouchEvent::TouchEnd(_)) => {
+                self.delay = None;
+                let now = Instant::now();
+                if self.loader.is_completely_grown(now) {
+                    return true;
+                }
+                if self.loader.is_animating() {
+                    self.loader.start_shrinking(ctx, now);
+                }
+            }
+            Event::Timer(token) if Some(token) == self.delay => {
+                self.delay = None;
+                self.pad.clear();
+                self.loader.start_growing(ctx, Instant::now());
+            }
+            _ => {}
+        }
+
+        match self.loader.event(ctx, event) {
+            Some(LoaderMsg::GrownCompletely) => {
+                // Wait for TouchEnd before returning.
+            }
+            Some(LoaderMsg::ShrunkCompletely) => {
+                self.loader.reset();
+                self.pad.clear();
+                ctx.request_paint()
+            }
+            None => {}
+        }
+
+        false
+    }
+}
+
+impl<T> Component for Homescreen<T>
+where
+    T: AsRef<str>,
+{
+    type Msg = HomescreenMsg;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        self.pad.place(AREA.translate(LOADER_OFFSET));
```

Comment:
seems wrong, probably the translated was meant to be used in loader

Reply:
d'oh, fixed [8dff0f7](https://github.com/trezor/trezor-firmware/pull/2538/commits/8dff0f7420698191712ea4e9a5e4b3bb7efba776)